addons/binding/org.openhab.binding.wifiled/src/main/java/org/openhab/binding/wifiled/handler/FadingWiFiLEDDriver.java

```diff
@@ -32,18 +39,17 @@
 
     public static final int DEFAULT_FADE_DURATION_IN_MS = 1000;
     public static final int DEFAULT_FADE_STEPS = 100;
+    private static final InternalLedState blackState = new InternalLedState();
 
     private boolean power = false;
-    private InternalLedState blackState = new InternalLedState();
-    private InternalLedState currentState = new InternalLedState();
-    private InternalLedState currentTargetState = new InternalLedState();
+    private InternalLedState currentState = new InternalLedState(); // USe to not update the controller with the same value
+    private InternalLedState _currentTargetState = new InternalLedState();
     private InternalLedState targetState = new InternalLedState();
-    private InternalLedState realTargetState = new InternalLedState();
     private LEDStateDTO dtoState = LEDStateDTO.valueOf(0, 0, 0, 0, 0, 0, 0, 0);
-    private ScheduledExecutorService executorService = Executors.newSingleThreadScheduledExecutor();
+    private ScheduledExecutorService executorService = Executors.newScheduledThreadPool(2);
+    private final Future<?> faderFuture[] = {CompletableFuture.completedFuture(null)};
```

Comment:
I needed to pass a reference to itself to the fader routine so it can cancel itself after the nuymber of fader steps was reached. I am planning to modify it to use AtomicReference like the below. I believe that's more clear and less error prone.

> private AtomicReference<Future<?>> faderReference = new AtomicReference<>(CompletableFuture.completedFuture(null));